ihatemoney/forms.py

```diff
@@ -182,13 +182,22 @@ class BillForm(FlaskForm):
     submit = SubmitField(_("Submit"))
     submit2 = SubmitField(_("Submit and add a new one"))
 
+    def __init__(self, bill_type=BillTypes.loan, *args, **kwargs):
+        super(BillForm, self).__init__(*args, **kwargs)
+        if bill_type in [BillTypes.loan, BillTypes.settlement]:
+            self.bill_type = bill_type
+        else:
+            self.bill_type = BillTypes.loan
```

Comment:
You don't need all this logic here. Just passing the default value ("loan" if we use a string) and we'll be good to go.